sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/ConstantFoldingSuite.scala

```diff
@@ -263,4 +263,19 @@ class ConstantFoldingSuite extends PlanTest {
 
     comparePlans(optimized, correctAnswer)
   }
+
+  test("Constant folding test with sideaffects") {
+    val originalQuery =
+      testRelation
+        .select("size(array(assert_true(false)))")
+
+    val optimized = Optimize.execute(originalQuery.analyze)
+
+    val correctAnswer =
+      testRelation
+        .select("size(array(assert_true(false)))")
+        .analyze
+
+    comparePlans(optimized, correctAnswer)
```

Comment:
This looks negative case, can we also add a positive case?

Reply:
those were covered in the other test, but I can add another one here as well

Reply:
I'm glad you pointed this out as it turns out I wasn't testing what I meant to in the negative case as well.

Reply:
Thanks for adding test.